File: python/2962. Count Subarrays Where Max Element Appears at Least K Times.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        total, max_element = (len(nums)*(len(nums)+1))//2, max(nums)
        left, count, answer = 0, 0, 0
        for right in range(len(nums)):
            if nums[right] == max_element:
                count += 1
            while count >= k:
                if nums[left] == max_element:
                    count -= 1
                left += 1
            answer += right - left + 1
        return total - answer

File: java/2962. Count Subarrays Where Max Element Appears at Least K Times.java
class Solution {
    public long countSubarrays(int[] nums, int k) {
        int max = Arrays.stream(nums).max().getAsInt();
        int n = nums.length;
        int count = 0;
        long answer;
        for (int left = 0, right = 0; right < nums.length; right++) {
            if (nums[right] == max)  count++;
            while (count >= k) {
                if (nums[left] == max) {
                    count--;
                }
                left++;
            }
            answer += left;
        }
        return answer;
    }
}